### PR TITLE
[DUOS-1161][risk=no] Fix user field validation on update

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1161

Fix the field validation when updating a user from json

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by raejohanek — COMMENTED** [review]
Solution looks good and makes sense, but I'm still getting the 400 response when I test this out locally through the UI, are you still getting that response too?

**Comment by rushtong** [comment]
@raejohanek - No, I don't. Are you sure you're pointing your local UI to your local instance of Consent? Check your configs.

File: src/main/java/org/broadinstitute/consent/http/service/ResearcherService.java [hunk 1]
```diff
@@ -166,12 +166,14 @@ private void validateRequiredFields(Map<String, String> properties) {
...
 
-    private void validateExistentFields(Map<String, String> properties) {
+    private Map<String, String> validateExistentFields(Map<String, String> properties) {
+        Map<String, String> newProps = new HashMap<>();
         properties.forEach((propertyKey, propertyValue) -> {
-            if (!UserFields.containsValue(propertyKey)) {
-                throw new IllegalArgumentException(propertyKey + " is not a valid property.");
+            if (UserFields.containsValue(propertyKey)) {
+                newProps.put(propertyKey, propertyValue);
             }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/ResearcherService.java:174]: There's no need to throw an exception, we just store the values we know and care about.

**Review by JVThomas — APPROVED** [review]
Nice fix! Profile update now goes through

**Comment by rushtong** [comment]
Thank you @JVThomas!

**Pull request merged**